### DISCUSSION TITLE
Is MDX format supported in v6?

### DISCUSSION THREAD

**Discussion by CezarCobuz** [storybookjs/react-native#293] [Q&A] [answered]
I am wondering if [MDX Format](https://storybook.js.org/docs/react/api/mdx) is currently supported and if the [addon-docs](https://storybook.js.org/addons/@storybook/addon-docs) can be seen otherwise as storybook server is not yet up in v6.

**Comment by dannyhw** [comment, marked as answer]
Hi Cezar, currently MDX is not supported ondevice. I'm not sure if there are any libraries out there to work with MDX in react native however that would be interesting to experiment with for a future ondevice addon. Open to all help also if anyone is able.

An alternative option would be the new package I'm working which makes it easy to use the reactjs storybook with react-native-web 
https://www.npmjs.com/package/@storybook/addon-react-native-web
I haven't tested much with MDX however the docs addon is already working with this react native web addon.
The package is kind of in a pre-alpha stage at the moment but an initial version should release soon. It's also possible to do without the addon you just need the right configuration. If you are interested in the react native web solution I can share more details.

**Reply by dannyhw** [reply]
Also I plan to support the server, it's just been a bit difficult to figure out how to update it for 6.0.

**Reply by CezarCobuz** [reply]
I can imagine. I wish I could help you with the development of this but I am still trying to get a grasp of the rn storybook concepts.

**Reply by dannyhw** [reply]
Asking questions and giving feedback is already valuable. Thanks for your interest in the project :)

**Reply by pke** [reply]
there is a pretty robust rn-markdown [component available ](https://github.com/iamacup/react-native-markdown-display) which has custom renderers which could be used to render RN components. In theory. But I guess it would need some kind of resolver to actually load the component to render and then interpret the MDX's params for that component.

**Reply by dannyhw** [reply]
The solutions I've seen for MDX require the user to setup mdx rendering on their project via some metro config and so on.

I think because of that it might instead be a case for a tutorial on how to set up MDX in react native storybook instead since I don't know if it's possible to build in that functionality to the library.

Happy for any suggestions or PRs though as always 😄
